src/components/popover-container/popover-container.pw.tsx

```diff
@@ -0,0 +1,683 @@
+
+    await expect(popoverContainerContent(page)).toHaveAttribute(
+      "aria-describedby",
+      testString
+    );
+  });
+
+  test("should render with openButtonAriaLabel", async ({ mount, page }) => {
+    await mount(<PopoverContainerComponent openButtonAriaLabel={testString} />);
+
+    const popoverContainerOpenComponent = getDataElementByValue(
+      page,
+      "popover-container-open-component"
+    );
+
+    await expect(popoverContainerOpenComponent).toHaveAttribute(
+      "aria-label",
+      testString
+    );
+  });
+
+  test("should render with closeButtonAriaLabel", async ({ mount, page }) => {
+    await mount(
+      <PopoverContainerComponent closeButtonAriaLabel={testString} />
+    );
+
+    const popoverCloseIconElement = getDataElementByValue(
+      page,
+      "popover-container-close-component"
+    );
+
+    await expect(popoverCloseIconElement).toHaveAttribute(
+      "aria-label",
+      testString
+    );
+  });
+
+  test("should render with containerAriaLabel", async ({ mount, page }) => {
+    await mount(<PopoverContainerComponent containerAriaLabel={testString} />);
+
+    const popoverContainerContentElement = popoverContainerContent(page);
+
+    await expect(popoverContainerContentElement).toHaveAttribute(
+      "aria-label",
+      testString
+    );
+  });
+
+  test("should render with renderOpenComponent", async ({ mount, page }) => {
+    await mount(<WithRenderOpenButtonComponent />);
+
+    const buttonComponent = getComponent(page, "button");
+    await buttonComponent.getByText("Test");
+
+    await expect(buttonComponent).toBeVisible();
+
+    const filterNewButton = getDataElementByValue(page, "filter_new");
+
+    await expect(filterNewButton).toBeVisible();
+
+    const popoverContainerContentElement = popoverContainerContent(page);
+
+    await expect(popoverContainerContentElement).not.toBeVisible();
+  });
+
+  test("should render with renderCloseComponent", async ({ mount, page }) => {
+    await mount(<WithRenderCloseButtonComponent />);
+
+    const getComponentElementButton = getComponent(page, "button");
+    await getComponentElementButton.getByText("Test");
+
+    await expect(getComponentElementButton).toBeVisible();
+
+    const popoverContainerContentElement = popoverContainerContent(page);
+
+    await expect(popoverContainerContentElement).toBeVisible();
+  });
+
+  const getBottomValue = (locator: Locator) =>
+    locator.evaluate((element) => element.getBoundingClientRect().bottom);
+
+  const getTopValue = (locator: Locator) =>
+    locator.evaluate((element) => element.getBoundingClientRect().top);
+
+  const getYValue = (locator: Locator) =>
+    locator.evaluate((element) => element.getBoundingClientRect().y);
+
+  ([
+    [true, 102, 178],
+    [false, 126, 202],
+  ] as const).forEach(([coverButton, yAndTopValueMin, bottomValueMin]) => {
+    test(`should render with shouldCoverButton prop set to ${coverButton}`, async ({
```

Comment:
**praise**: well done for rewriting the flaky test in a way that's easy to read and (I assume!) actually works

Reply:
Thanks. Well, time will tell on if it actually works 😆